nanodbc/nanodbc.cpp

```diff
@@ -410,8 +410,13 @@ recent_error(SQLHANDLE handle, SQLSMALLINT handle_type, long& native, std::strin
     } while (rc != SQL_NO_DATA);
 
     convert(std::move(result), rvalue);
-    if (size(sql_state) > 0)
-        state = std::string(&sql_state[0], &sql_state[size(sql_state) - 1]);
+    if (size(sql_state) > 0) {
+        state.clear();
+        for (int idx = 0; idx != size(sql_state) - 1; ++idx) {
```

Comment:
Could you replace `int` with `std::size_t`? Since `sql_state` is constant size array, the condition is not prone to unsigned wrapping.

That would help to clear this warning-as-error
```
error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
         for (int idx = 0; idx != size(sql_state) - 1; ++idx) 
```

Reply:
Right, it wouldn't be prone either way, fixed!